Automattic/jetpack
# AI Featured Image: add media source entry point for the tool

Fixes #37165.

## Proposed changes:

* Add support for "featured image exclusive" media sources on the media source dropdown, to include options available only on the featured image menu
* Add "AI Generated Image" option on the media source dropdown, as a "featured image exclusive" source
* Refactor the `FeaturedImage` component so it can be used on different placements, not only on the Jetpack sidebar
* Make the new media source option shows the `FeaturedImage` component
* Make the `FeaturedImage` component triggers the first image generation when it's used on the "media source" menu

Expected result:

<img width="250" alt="Screenshot 2024-05-01 at 16 39 41" src="https://github.com/Automattic/jetpack/assets/6760046/3ba80bdb-bc8c-472f-bb55-1e69eaaabe36">

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor and add some content to it
* Open the browser console and enable the debug messages by running `localStorage.setItem( 'debug', '*' );`
* On the "Settings" sidebar (the default one), look for the featured image section
   * it can be on the Post Status section, or can be a specific section on the sidebar, depending on the version of Gutenberg you are running
* Click "Set featured image"; a dropdown with media sources will show up, and a "AI Generated Image" option is available:

<img width="250" alt="Screenshot 2024-05-01 at 16 39 41" src="https://github.com/Automattic/jetpack/assets/6760046/3ba80bdb-bc8c-472f-bb55-1e69eaaabe36">

* Click "AI Generated Image"; the Jetpack AI Featured Image Generator modal will show up, on a loading state
* Confirm a new image is generated and visible on the modal
* Confirm you can use the tool in the same way it can be used when triggered from the Jetpack sidebar (generate new images, use custom prompt, set an image as featured)
* Confirm the tracking events are generated with `placement = media-source-dropdown`:

<img width="500" alt="Screenshot 2024-05-01 at 16 39 08" src="https://github.com/Automattic/jetpack/assets/6760046/70a9564f-83d7-4c67-aeba-b74d3edf1bf6">

* Confirm the tool is **not available** on the dropdown for other locations, like the image block:

<img width="500" alt="Screenshot 2024-05-01 at 16 51 58" src="https://github.com/Automattic/jetpack/assets/6760046/5aaafa28-5081-4e86-ab64-245d47123ff1">
